### PR TITLE
Implement fetch sponsors

### PR DESCRIPTION
**Pull request by vickiekamau**
# Scope

_Please make sure to read https://github.com/droidconKE/droidconKE2022Android/docs/CONTRIBUTING.md
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [x] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
Closed the following issue on Trello https://trello.com/c/OYs0ZDnN/38-data-fetch-sponsors-from-api

## Other testing QA Notes
Have testes Fetch Sponsor API

Please add a screenshot (if necessary)

### CODE REVIEW

File: data/build.gradle.kts [hunk 1]
```diff
@@ -57,6 +57,7 @@ dependencies {
     api(libs.kotlin.coroutines.datetime)
     implementation(libs.android.hilt)
     implementation(libs.timber)
+    implementation("androidx.paging:paging-common-ktx:3.1.1")
```
Comment (wangerekaharun) [inline data/build.gradle.kts:60]: Kindly check the read me on how to add dependencies and correct this

File: data/build.gradle.kts [hunk 2]
```diff
@@ -70,6 +71,9 @@ dependencies {
     androidTestImplementation(libs.android.test.junit4)
     androidTestImplementation(libs.android.test.espresso)
 
+    // paging
```
Comment (wangerekaharun) [inline data/build.gradle.kts:74]: Comments not necessary on libs

**Review by wangerekaharun — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import javax.inject.Inject
+
+
+class SponsorsApi @Inject constructor(
+    private val client: HttpClient
+) {
+
+    suspend fun fetchSponsors(perPage: Int, page: Int): DataResult<SponsorsPagedResponse> =
+       dataResultSafeApiCall{
+            val response = client.get("${Constants.BASE_URL}/events/droidconke-2022-797/feeds?per_page=$perPage&page=$page")
```
Comment (jumaallan) [inline data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt:35]: Can you inline this?

```suggestion
           client.get("${Constants.BASE_URL}/events/droidconke-2022-797/feeds?per_page=$perPage&page=$page").body()
```

File: data/build.gradle.kts (hunk 1 above)
Comment (vickiekamau) [reply data/build.gradle.kts]: I have updated the dependency according to the read me guidelines

File: data/build.gradle.kts (hunk 2 above)
Comment (vickiekamau) [reply data/build.gradle.kts]: I have removed the comment

File: data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt (hunk 3 above)
Comment (vickiekamau) [reply data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt]: This has been updated

File: data/build.gradle.kts (hunk 1 above)
Comment (wangerekaharun) [reply data/build.gradle.kts]: Looks good

**Review by wangerekaharun — APPROVED** [review]

**Comment by wangerekaharun** [comment]
@vickiekamau Resolve conflicts now

**Comment by vickiekamau** [comment]
> @vickiekamau Resolve conflicts now

@wangerekaharun this has been resolved

**Comment by wangerekaharun** [comment]
@vickiekamau Seen, thanks!

File: data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt (hunk 3 above)
Comment (wangerekaharun) [reply data/src/main/java/com/android254/data/network/apis/SponsorsApi.kt]: @jumaallan Can you check if this is now resolved?

**Review by misshannah — APPROVED** [review]
Cool stuff!

**Review by wangerekaharun — CHANGES_REQUESTED** [review]
@vickiekamau Kindly remove the pagination stuff. Since the data from API will not be that large we agreed to do away with pagination

**Review by kibettheophilus — APPROVED** [review]

**Comment by vickiekamau** [comment]
> @vickiekamau Kindly remove the pagination stuff. Since the data from API will not be that large we agreed to do away with pagination

@wangerekaharun kindly check pagination has been removed

**Review by wangerekaharun — APPROVED** [review]

**Comment by wangerekaharun** [comment]
> > @vickiekamau Kindly remove the pagination stuff. Since the data from API will not be that large we agreed to do away with pagination
> 
> @wangerekaharun kindly check pagination has been removed

Looks good!

**Pull request merged**